### PR TITLE
refactor(chore): fix sonar code smells

### LINKED ISSUE #204: Fix All the sonar issues
**Describe the bug**
Fix overall all the sonar issues 
https://sonarcloud.io/summary/overall?id=sourcefuse_loopback4-authentication

### PR DESCRIPTION
**Pull request by sf-sahil-jassal**
fix sonar code smells related to function type,
any typings and empty constructors

GH-204

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #204 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -8,6 +8,9 @@ import {Request, Response} from '@loopback/rest';
 import {VerifyFunction} from './strategies';
 export * from './strategies/types';
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type AnyObject = Record<string, any>; // NOSONAR
+
```
Comment (yeshamavani) [inline src/types.ts:13]: instead of this use @loopback/repository

Comment (yeshamavani) [reply src/types.ts:13]: no need to create your own type 
this will change the imports everywhere

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 1 above)
Comment (sf-sahil-jassal) [reply src/types.ts:13]: ok sure

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**